test/utils/cluster.go

```diff
@@ -0,0 +1,123 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const listFormat = "value(NAME)"
+
+func EnsureCluster(clusterName, zone string, numOfNodes int) error {
+	if !isClusterExisting(clusterName, zone) {
+		klog.Infof("Cluster %s in zone %s does not exist, creating.", clusterName, zone)
+		return createCluster(clusterName, zone, numOfNodes)
+	}
+	params := []string{
+		"container",
+		"clusters",
+		"describe",
+		clusterName,
+		"--zone", zone,
+	}
+	out, err := exec.Command("gcloud", params...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("cannot describe cluster %s using gcloud: %w", clusterName, err)
+	}
+	if err := verifyClusterInfo(string(out), numOfNodes); err != nil {
+		return fmt.Errorf("cluster %s does not match the given config: %w.", clusterName, err)
+	}
+	klog.Infof("Use existing cluster %s in zone %s with %d nodes", clusterName, zone, numOfNodes)
+	return nil
+}
+
+func createCluster(clusterName, zone string, numOfNodes int) error {
+	params := []string{
+		"container",
+		"clusters",
+		"create",
+		clusterName,
+		"--zone", zone,
+		"--num-nodes", strconv.Itoa(numOfNodes),
+	}
+	if out, err := exec.Command("gcloud", params...).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed creating cluster: %s, err: %w", out, err)
+	}
+	return nil
+}
+
+func DeleteCluster(clusterName, zone string) error {
+	params := []string{
+		"container",
+		"clusters",
+		"delete",
+		clusterName,
+		"--zone", zone,
+		"--quiet",
+	}
+	if out, err := exec.Command("gcloud", params...).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed deleting cluster: %s, err: %w", out, err)
+	}
+	return nil
+}
+
+// GetCredentials will updates the user's cluster credentials that are saved
+// in their ~/.kube/config directory and switch to the provided cluster.
+func GetCredential(clusterName, zone string) error {
+	params := []string{
+		"container",
+		"clusters",
+		"get-credentials",
+		clusterName,
+		"--zone", zone,
+	}
+	if out, err := exec.Command("gcloud", params...).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed setting kubeconfig: %s, err: %w", out, err)
+	}
+	return nil
+}
+
+// isClusterExisting checks if the given cluster exists in the given zone.
+func isClusterExisting(clusterName, zone string) bool {
+	clusters := listClusters(zone)
+	for _, c := range clusters {
+		if c == clusterName {
+			return true
+		}
+	}
+	return false
+}
+
+// listClusters lists cluster names in the given zone with Value format.
+func listClusters(zone string) []string {
+	params := []string{
+		"container",
+		"clusters",
+		"list",
+		"--zone", zone,
+		"--format", listFormat,
```

Comment:
It does not make sense to make listFormat a module level variable. This is something that only matters to the implementation inside listClusters.

Reply:
Just take the string and replace the const with it here inline.

Reply:
Updated.